### PR TITLE
Fix for 3rd Party Clients copy button

### PR DESCRIPTION
**Pull request by therobbiedavis**
# Fixed
- Fixed: Fixed an issue where the copy button wasn't working on the "3rd Party Clients" page.

### CODE REVIEW

File: UI/Web/src/app/user-settings/api-key/api-key.component.ts [hunk 1]
```diff
@@ -43,8 +44,10 @@ export class ApiKeyComponent implements OnInit, OnDestroy {
     this.onDestroy.complete();
   }
 
-  async copy() {
-    await navigator.clipboard.writeText(this.key);
+  async copy(input: any) {
```
Comment (majora2007) [inline UI/Web/src/app/user-settings/api-key/api-key.component.ts:47]: First, don't use any types, this should be a string. Second, why are you passing it from the template and not just referencing this.key directly?

Comment (majora2007) [reply UI/Web/src/app/user-settings/api-key/api-key.component.ts:47]: Reading the copy code vs how it's invoked, I'm confused. It's supposed to be an HTMLInputElement type, but not sure how the template is getting a reference to this. 

I think the mode ideal way would be to setup a ViewChild to reference the input:

@ViewChild('keyInput') inputElem: ElementRef<HTMLInputElement>;

Then in copy, use this.inputElem.nativeElement to use the .select and setSelectionRange, etc.

**Review by majora2007 — CHANGES_REQUESTED** [review]

File: UI/Web/src/app/user-settings/api-key/api-key.component.ts (hunk 1 above)
Comment (therobbiedavis) [reply UI/Web/src/app/user-settings/api-key/api-key.component.ts]: Resolved in 9692ab3

File: UI/Web/src/app/user-settings/api-key/api-key.component.html [hunk 2]
```diff
@@ -4,7 +4,7 @@
     <div class="input-group">
         <input #apiKey type="text" readonly class="form-control" id="api-key--{{title}}" aria-describedby="button-addon4" [value]="key" (click)="selectAll()">
         <div class="input-group-append" id="button-addon4">
-            <button class="btn btn-outline-secondary" type="button" (click)="copy()"><span class="sr-only">Copy</span><i class="fa fa-copy" aria-hidden="true"></i></button>
+            <button class="btn btn-outline-secondary" type="button" (click)=copy()><span class="sr-only">Copy</span><i class="fa fa-copy" aria-hidden="true"></i></button>
```
Comment (majora2007) [inline UI/Web/src/app/user-settings/api-key/api-key.component.html:7]: Looks like you're missing the "" around the copy call. Please add them in.

**Review by majora2007 — CHANGES_REQUESTED** [review]

File: UI/Web/src/app/user-settings/api-key/api-key.component.html (hunk 2 above)
Comment (therobbiedavis) [reply UI/Web/src/app/user-settings/api-key/api-key.component.html]: Resolved in 8922be2

**Review by majora2007 — APPROVED** [review]

**Pull request merged**